src/AutoMapper/Mappers/ExpressionMapper.cs

```diff
@@ -247,7 +247,13 @@ private Type GetSourceType(PropertyMap propertyMap)
             {
                 if (propertyMap.SourceMember == null)
                 {
-                    return propertyMap.SourceType;
+                    return propertyMap.SourceType ??
```

Comment:
When it's mapped through a resolver it returns typeof(object) (I think I put that one in there :)). Probably it can be null only when it's not mapped.